cmd/rancher-desktop-guestagent/main.go

```diff
@@ -75,29 +80,60 @@ func main() {
 
 	group, ctx := errgroup.WithContext(context.Background())
 
-	if *enableDocker {
+	if *enablePriviledService && *enableIptables {
+		log.Fatal("error enabling Privileged Service mode and iptables, you can only enable one.")
+	}
+
+	if *enablePriviledService {
+		if !*enableContainerd && !*enableDocker {
+			log.Fatal("enabling Privileged Service mode requires either Docker or Containerd enabled.")
+		}
+
+		if *enableContainerd && *enableDocker {
+			log.Fatal("enabling Privileged Service mode requires enabling either Docker or Containerd not both.")
+		}
+
 		if *vtunnelAddr == "" {
-			log.Fatal("vtunnel address must be provided when docker is enable.")
+			log.Fatal("vtunnel address must be provided when docker is enabled.")
 		}
 
-		group.Go(func() error {
-			wslAddr, err := getWSLAddr(wslInfName)
-			if err != nil {
-				return err
-			}
-			forwarder := forwarder.NewVtunnelForwarder(*vtunnelAddr)
-			portTracker := tracker.NewPortTracker(forwarder, wslAddr)
-			eventMonitor, err := docker.NewEventMonitor(portTracker)
-			if err != nil {
-				return fmt.Errorf("error initializing docker event monitor: %w", err)
-			}
-			if err := tryConnectDocker(ctx, eventMonitor.Info); err != nil {
-				return err
-			}
-			eventMonitor.MonitorPorts(ctx)
+		wslAddr, err := getWSLAddr(wslInfName)
+		if err != nil {
+			log.Fatalf("gettting wsl IP addrs: %w", err)
+		}
 
-			return nil
-		})
+		forwarder := forwarder.NewVtunnelForwarder(*vtunnelAddr)
+		portTracker := tracker.NewPortTracker(forwarder, wslAddr)
+
+		if *enableContainerd {
+			group.Go(func() error {
+				eventMonitor, err := containerd.NewEventMonitor(*containerdSock, portTracker)
+				if err != nil {
+					return fmt.Errorf("error initializing containerd event monitor: %w", err)
+				}
+				if err := tryConnectAPI(ctx, containerdSocketFile, eventMonitor.IsServing); err != nil {
+					return err
+				}
+				eventMonitor.MonitorPorts(ctx)
+
+				return eventMonitor.Close()
+			})
+		}
+
+		if *enableDocker {
+			group.Go(func() error {
+				eventMonitor, err := docker.NewEventMonitor(portTracker)
+				if err != nil {
+					return fmt.Errorf("error initializing docker event monitor: %w", err)
+				}
+				if err := tryConnectAPI(ctx, dockerSocketFile, eventMonitor.Info); err != nil {
+					return err
+				}
+				eventMonitor.MonitorPorts(ctx)
+
+				return nil
+			})
+		}
 	}
 
 	tracker := tcplistener.NewListenerTracker()
```

Comment:
Maybe rename this to `tcpTracker`, since there's now different trackers in this function?

Reply:
I have a follow up PR that renames that.